pkg/controller/controller_ref_manager.go

```diff
@@ -107,6 +107,12 @@ func (m *BaseControllerRefManager) ClaimObject(ctx context.Context, obj metav1.O
 		// Ignore if the object is being deleted
 		return false, nil
 	}
+
+	if len(m.Controller.GetNamespace()) > 0 && m.Controller.GetNamespace() != obj.GetNamespace() {
```

Comment:
is there a unit test exercising adoption of a namespaced pod by a cluster-scoped controller?